pineappl/src/grid.rs

```diff
@@ -1482,6 +1483,21 @@ impl Grid {
             })
             .collect();
     }
+
+    /// Merges the factors of the channels into the subgrids to normalize channel coefficients.
+    ///
+    /// This method factors out the smallest absolute coefficient from each channel using
+    /// [`boc::Channel::factor`] and then scales the corresponding subgrids by these factors.
+    pub fn merge_channel_factors(&mut self) {
```

Comment:
Adding a unit test for this should be trivial as the action is supposed to be an identity (on the convolution numbers), right?

Reply:
What exactly do you mean here (w.r.t. the identity)?

Reply:
take a (already existing) unit test for calling `convolute_with_one`:
```rust
pub fn test_convolute_with_one() {
 // create grid ...
 // call convolute_with_one ...
 // check numbers ...
 // ADD HERE: call merge_channel_factors
 // ADD HERE: check the numbers are still the same - i.e. identical to above
}
```
something like this should work, no?

Reply:
Oh, yes, ofc. This operation is invariant under the convolution. We no longer have unit tests with `convolute_with_one` and this can only really be checked at the level of the Python API, so I added some in [f07d9f0](https://github.com/NNPDF/pineappl/pull/353/commits/f07d9f03f266f721d5a6bdaf125985027f886d61).